File: src/main/java/step1/Movie.java
package step1;

public class Movie {
	public static final int CHILDRENS = 2;
	public static final int REGULAR = 0;
	public static final int NEW_RELEASE = 1;
	
	private String title;
	private int priceCode;

	public Movie(String title, int priceCode) {
		this.title = title;
		this.priceCode = priceCode;
	}

	public int getPriceCode() {
		return priceCode;
	}

	public void setPriceCode(int arg) {
		priceCode = arg;
	}

	public String getTitle() {
		return title;
	};

	double amountFor(Rental rental) {
		// determine amounts for each line

		double thisAmount = 0;
		switch (rental.getMovie().getPriceCode()) {
        case REGULAR:
            thisAmount += 2;
            if (rental.getDaysRented() > 2)
                thisAmount += (rental.getDaysRented() - 2) * 1.5;
            break;

        case NEW_RELEASE:
            thisAmount += rental.getDaysRented() * 3;
            break;

        case CHILDRENS:
            thisAmount += 1.5;
            if (rental.getDaysRented() > 3)
                thisAmount += (rental.getDaysRented() - 3) * 1.5;
            break;
        }
		return thisAmount;
	}
}

File: src/main/java/step1/Movie.kt
package step1

class Movie(val title: String, var priceCode: Int) {
    internal val DEFAULT_POINTS = 1

    internal fun amountFor(daysRented: Int): Double {
        var thisAmount = 0.0

        when (priceCode) {
            REGULAR -> {
                thisAmount += 2.0
                if (daysRented > 2)
                    thisAmount += (daysRented - 2) * 1.5
            }

            NEW_RELEASE -> thisAmount += (daysRented * 3).toDouble()

            CHILDRENS -> {
                thisAmount += 1.5
                if (daysRented > 3)
                    thisAmount += (daysRented - 3) * 1.5
            }
        }
        return thisAmount
    }

    internal fun getFrequentRenterPoints(daysRented: Int): Int {
        return DEFAULT_POINTS + getBonusPoints(daysRented)
    }

    private fun getBonusPoints(daysRented: Int): Int {
        return if (priceCode == NEW_RELEASE && daysRented > 1) 1 else 0
    }

    companion object {
        val CHILDRENS = 2
        val REGULAR = 0
        val NEW_RELEASE = 1
    }
}